### ISSUE TITLE
DC2 benchmarking

### ISSUE THREAD

**Issue by jeff-regier** [prob-ml/bliss#726]
A description of DC2: https://arxiv.org/abs/2010.05926
Data access: https://data.lsstdesc.org/doc/dc2_sim_sky_survey

Steps:

- [x] Create a class called `Dc2FullCatalog` in `surveys/lsst.py` that inherits from `FullCatalog` and loads the true DC2 catalog and provides access to it. It should be analogous to `PhotoFullCatalog` and `DecalsFullCatalog`, and be exercised by a pytest test.
- [x] Create a class called `DC2` in `surveys/lsst.py` that inherits from [`pl.LightningDataModule`](https://lightning.ai/docs/pytorch/stable/data/datamodule.html), and implements methods such as `train_dataloader`, `val_dataloader()`, and `test_dataloader()`. These data loaders should provide both images and the true catalog corresponding to each image to the encoder.
- [x] rebase on master and change the `DC2` class to inherit from `Survey` (like `DarkEnergyCameraLegacySurvey` does now)
- [x] train encoder with DC2 data and verify that training loss decreases consistently across epochs
- [x] verify the validation loss decreases with training, that validation metrics (especially f1 score) improve with training, and the validation metrics reach a nontrivial level of performance (i.e., detecting most bright light sources)
